src/components/CodeInput/CodeInput.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {ReactNode, useCallback, useRef, useState, memo} from 'react'
+import {TextInput, Text, StyleProp, ViewStyle, KeyboardTypeOptions, TextStyle} from 'react-native'
+import {metrics} from '../../helpers/metrics'
+import styled from 'styled-components/native'
+import Cursor from './Cursor'
+import type {ITheme} from 'src/theme'
+
+interface CodeInputProps {
+  /** define style for cell */
+  cellStyle?: StyleProp<ViewStyle>
+
+  /** define style for cell when cell is focused */
+  focusCellStyle?: StyleProp<ViewStyle>
+
+  /** define style for text in the cell */
+  textStyle?: StyleProp<TextStyle>
+
+  /** define secure view style when using secureTextEntry mode */
+  secureViewStyle?: StyleProp<ViewStyle>
+
+  /** define style for text in the cell when cell is focused */
+  focusTextStyle?: StyleProp<TextStyle>
+
+  /** cell count */
+  length?: number
+  /** callback when complete  */
+
+  onSubmit?: (val: string) => void
+
+  /** render custom view for cursor/indicator */
+  customCursor?: () => ReactNode
+
+  /** render custom view for cursor/indicator */
+  secureTextEntry?: boolean
+
+  /** keyboard type */
+  keyboardType?: KeyboardTypeOptions
```

Comment:
could we have an props like a callback and that callback will be fire when inputs is fulfill ?

Reply:
Yes. that is the onSubmit prop, will fire when input is fulfill